### ISSUE TITLE
Exception thrown by setScan method since upgrade to reflections library 0.9.11 using a custom classloader

### ISSUE THREAD

**Issue by vincentmigot** [swagger-api/swagger-core#3295]
**Description**
I use a custom classloader for a custom plugin system.
Those plugins can contain Swagger API definition.
In my main application I use the BeanConfig.setScan(true) method to initialize swagger.
Since I upgrade swagger-jersey2-jaxrs from version 1.5.13 to 1.5.23 I get the following stack trace exception.
I tried all version from 1.5.14 to 1.5.23 and all produce the same issue.
So I'm pretty sure that this exception is caused by the upgrade of reflections library from version 0.9.10 to 0.9.11.

```
org.reflections.ReflectionsException: could not get type for name javax.jms.Connection
	at org.reflections.ReflectionUtils.forName(ReflectionUtils.java:390)
	at org.reflections.Reflections.expandSuperTypes(Reflections.java:381)
	at org.reflections.Reflections.<init>(Reflections.java:126)
	at io.swagger.jaxrs.config.BeanConfig.classes(BeanConfig.java:276)
	at io.swagger.jaxrs.listing.BaseApiListingResource.scan(BaseApiListingResource.java:59)
	at io.swagger.jaxrs.listing.BaseApiListingResource.process(BaseApiListingResource.java:120)
	at io.swagger.jaxrs.listing.BaseApiListingResource.getListingJsonResponse(BaseApiListingResource.java:155)
	at io.swagger.jaxrs.listing.ApiListingResource.getListing(ApiListingResource.java:32)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.glassfish.jersey.server.model.internal.ResourceMethodInvocationHandlerFactory.lambda$static$0(ResourceMethodInvocationHandlerFactory.java:52)
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher$1.run(AbstractJavaResourceMethodDispatcher.java:124)
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.invoke(AbstractJavaResourceMethodDispatcher.java:167)
	at org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$ResponseOutInvoker.doDispatch(JavaResourceMethodDispatcherProvider.java:176)
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.dispatch(AbstractJavaResourceMethodDispatcher.java:79)
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:469)
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:391)
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:80)
	at org.glassfish.jersey.server.ServerRuntime$1.run(ServerRuntime.java:253)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:248)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:244)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:292)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:274)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:244)
	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:265)
	at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:232)
	at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:680)
	at org.glassfish.jersey.servlet.WebComponent.serviceImpl(WebComponent.java:392)
	at org.glassfish.jersey.servlet.WebComponent.service(WebComponent.java:346)
	at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:365)
	at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:318)
	at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:205)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:202)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96)
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:526)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:139)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343)
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:408)
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66)
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:860)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1589)
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.ClassNotFoundException: javax.jms.Connection
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:583)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
	at org.reflections.ReflectionUtils.forName(ReflectionUtils.java:388)
	... 54 common frames omitted
```

This issue is kind of related to this one: https://github.com/swagger-api/swagger-core/issues/2597
The workaround defined in it work for it (excluding reflections 0.9.11 and explicitly add dependency to 0.9.10)

**Why it's a regression in my opinion**
In my opinion it's clearly a regression in swagger due to this library upgrade:
1. As you can see in the stack the Exception come from Reflections.expandSuperTypes method which has been added in version 0.9.11(https://github.com/ronmamo/reflections/commit/ad55f7fdd7c54cd0b583ca275748e3428ece5b96 and https://github.com/ronmamo/reflections/commit/22d1ddc311d0498a07d656258d12124bd42d28f9)

2. In my case this method call the "forName" method without using my custom classloader and fail but there is no way to pass it through

3. This change breaks swagger behavior, as all was working perfectly fine before.

4. This new behavior of reflections library can be configured through "shouldExpandSuperTypes" configuration option which is true by default, set it to false by default won't have break swagger previous behavior.

**Solutions**
Appart for the trivial non-solution of going back to version 0.9.10, here are my thougths:
- Set shouldExpandSuperTypes option to false in swagger : this may not be the best option because this functionality can be useful for others and is the new behavior of swagger since version 1.5.14
- Allow user to define this shouldExpandSuperTypes option through swagger API: my favorite !

If you agreed with any of this solutions or have a better one, I can probably found some time to develop it.

Thanks for your great work !

**Comment by vincentmigot** [comment]
In fact as my thought were purely theoretical, I actually implement the second solution in this pull request https://github.com/swagger-api/swagger-core/pull/3296

It solve my problem, confirming my analysis.

**Comment by frantuma** [comment]
Thanks a lot for the investigation and PR! Merged #3296
